lib/cfg-lexer.c

```diff
@@ -1273,6 +1273,15 @@ cfg_token_block_free(CfgTokenBlock *self)
   g_free(self);
 }
 
+void
+cfg_lexer_track_file(GlobalConfig *self, const gchar *file_path, const gchar *path_type)
```

Comment:
I like having this structure, and the functions that supports this type, but those functions scattered across source files via convenience; which should not drive the design. Like this function has not much to do with the lexer, even it uses the `GlobalConfig` as `self` (the original name suggested is better).

As such all of the structure and things (functions connected to this type and structure, typedef alias) you should either move into the same file as `GlobalConfig` (which is well not the best option but better imho than storing in lexer), or you could create its own header/source file and include where it is needed (which I would enforce when it became more complex than the current, now just one suggestion).

Reply:
@Kokan : Yes, I have created a new source file called ```cfg-path.c``` now.